build.rs

```diff
@@ -75,63 +105,67 @@ fn get_value_from_table<'a>(table: &'a Value, key: &str) -> Result<&'a Value> {
 // parameters to the setup_mbed_crypto.sh script which clones and builds Mbed Crypto and create
 // a static library.
 fn setup_mbed_crypto(mbed_config: &MbedConfig, mbed_version: &str) -> Result<()> {
-    let mut run_script = ::std::process::Command::new(SETUP_MBED_SCRIPT_PATH);
-    run_script.arg(mbed_version).arg(
-        mbed_config
-            .mbed_path
-            .clone()
-            .unwrap_or(String::from(env::var("OUT_DIR").unwrap())),
-    );
-
-    let toolchain;
-    let mbed_compiler;
-    let mbed_archiver;
-    if std::env::var("TARGET").unwrap() == "aarch64-unknown-linux-gnu" {
-        toolchain = mbed_config
-            .aarch64_unknown_linux_gnu
-            .as_ref()
-            .ok_or_else(|| {
+    let (mbed_compiler, mbed_archiver) =
+        if std::env::var("TARGET").unwrap() == "aarch64-unknown-linux-gnu" {
+            let toolchain;
+            toolchain = mbed_config
+                .aarch64_unknown_linux_gnu
+                .as_ref()
+                .ok_or_else(|| {
+                    Error::new(
+                        ErrorKind::InvalidInput,
+                        "The aarch64_unknown_linux_gnu subtable of mbed_config should exist",
+                    )
+                })?;
+            (
+                toolchain
+                    .mbed_compiler
+                    .clone()
+                    .unwrap_or_else(|| DEFAULT_ARM64_MBED_COMPILER.to_string()),
+                toolchain
+                    .mbed_archiver
+                    .clone()
+                    .unwrap_or_else(|| DEFAULT_ARM64_MBED_ARCHIVER.to_string()),
+            )
+        } else {
+            let toolchain;
+            toolchain = mbed_config.native.as_ref().ok_or_else(|| {
```

Comment:
Do you think we should keep the subtable as required but the keys in it as optional? The other way around sounds more sensible - put a subtable in your `build-config.toml` with the correct keys if you want to change smth

Reply:
I don't really mind, maybe that could be discussed as part of #62 ?

Reply:
Sure!